src/clr-angular/utils/drag-and-drop/providers/drag-handle-registrar.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, Renderer2} from "@angular/core";
+
+import {ClrDragEventListener} from "./drag-event-listener";
+
+// This provider registers the drag handle element.
+// When it registers a element as a drag handle, it attaches that element to the listeners from ClrDragEventListener.
+// Also, it adds the "drag-handle" css class to the registered element through Renderer.
+@Injectable()
+export class ClrDragHandleRegistrar<T> {
+    private _customHandleEl: Node;
+    private _defaultHandleEl: Node;
+
+    set defaultHandleEl(value: Node) {
+        this._defaultHandleEl = value;  // defaultHandleEl will be usually the clrDraggable element.
+
+        // If the customHandleEl has been registered,
+        // don't make the defaultHandleEl the drag handle yet until the customHandleEl is unregistered.
+        if (!this._customHandleEl) {
+            this.makeElementHandle(this._defaultHandleEl);
+        }
+    }
+
+    constructor(private dragEventListener: ClrDragEventListener<T>, private renderer: Renderer2) {}
+
+    private makeElementHandle(el: Node) {
+        if (this._defaultHandleEl && this._defaultHandleEl !== el) {
+            // Before making an element the custom handle element,
+            // we should remove the existing drag-handle class from the draggable element.
+            this.renderer.removeClass(this._defaultHandleEl, "drag-handle");
+        }
+        this.dragEventListener.attachDragListeners(el);
+        this.renderer.addClass(el, "drag-handle");
+    }
+
+    get customHandle() {
+        return this._customHandleEl;
+    }
+
+    public registerCustomHandle(handleElement: Node) {
+        this.dragEventListener.detachDragListeners();  // removes the listeners only when there are existing ones
+        this._customHandleEl = handleElement;
+        this.makeElementHandle(this._customHandleEl);
+    }
+
+    public unregisterCustomHandle() {
+        delete this._customHandleEl;
+        // if default handle is set, make that handle
```

Comment:
You're missing the `removeClass()` and `detachDragListeners()`.

Reply:
Added in and called the methods.